### PR TITLE
Feature/#923 create taipy common

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**
Add taipy-common package. This package will contain taipy-config, _cli and logger.

### CODE REVIEW

File: taipy/__init__.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 if find_spec("taipy"):
     if find_spec("taipy.config"):
```
Comment (jrobinAV) [inline taipy/__init__.py]: SHould this be changed as well?

the package is taipy.common, no?

File: taipy/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
...
-        from taipy.config._init import *
+        from taipy.common.config._init import *
```
Comment (jrobinAV) [inline taipy/__init__.py:16]: Should we move the _init to taipy.common?
from taipy.common._init import *

File: taipy/common/config/__init__.py [hunk 3]
```diff
@@ -21,9 +21,9 @@
 !!! example "Standard usage"
```
Comment (jrobinAV) [inline taipy/common/config/__init__.py:19]: The previous lines should be changed as well replacing `taipy.config` by `taipy.common.config`

File: taipy/common/config/_serializer/_base_serializer.py [hunk 4]
```diff
@@ -101,9 +101,7 @@ def _from_dict(cls, as_dict) -> _Config:
         for section_name, sect_as_dict in as_dict.items():
             if section_class := cls._section_class.get(section_name, None):
                 if issubclass(section_class, UniqueSection):
-                    config._unique_sections[section_name] = section_class._from_dict(
-                        sect_as_dict, None, None
-                    )  # type: ignore
+                    config._unique_sections[section_name] = section_class._from_dict(sect_as_dict, None, None)  # type: ignore
```
Comment (jrobinAV) [inline taipy/common/config/_serializer/_base_serializer.py]: I believe it is over 120 chars.

**Review by jrobinAV — COMMENTED** [review]
Seems right.

File: .github/workflows/partial-tests.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ jobs:
 
       - uses: jpetrucciani/mypy-check@master
         with:
-          mypy_flags:  "--ignore-missing-imports --implicit-optional --no-namespace-packages --exclude (taipy/templates/|tools/|doc/gui/examples/.*/builder.py|taipy/common/config/config.pyi) --follow-imports skip --disable-error-code import-untyped"
+          mypy_flags:  "--ignore-missing-imports --implicit-optional --disable-error-code attr-defined --no-namespace-packages --exclude (taipy/templates/|tools/|doc/gui/examples/.*/builder.py|taipy/common/config/config.pyi) --follow-imports skip --disable-error-code import-untyped"
```
Comment (jrobinAV) [inline .github/workflows/partial-tests.yml:17]: IS it a permanent setting or just temporary?

**Review by jrobinAV — DISMISSED** [review]
It looks good to me. But only a dev release will make me fully confident it's working well.

File: .github/workflows/partial-tests.yml (hunk 5 above)
Comment (joaoandre-avaiga) [reply .github/workflows/partial-tests.yml:17]: Probably temporary. We have a lot of lines we atributes not defined in config due to injecting code from core and gui. I believe we can revert after merging this PR

**Review by jrobinAV — APPROVED** [review]

**Review by trgiangdo — APPROVED** [review]

**Comment by jrobinAV** [comment]
I installed the package, and it seems right.

**Pull request merged**